### PR TITLE
Initial Setup

### PR DESCRIPTION
**Pull request by rickycodes**
This fully sets up the repo as a functioning GitHub action that publishes an npm module.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: 'NPM Publish'
+description: 'NPM Publish the release'
```
Comment (rekmarks) [inline action.yml:2]: They lowercase the name in their branding: https://www.npmjs.com/about

```suggestion
name: 'Publish to npm'
description: 'Publish the release to npm'
```

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+inputs:
+  npm-token:
+    description: 'The token used for npm publishing. Publishing to npm will be skipped if omitted.'
```
Comment (rekmarks) [inline action.yml:5]: With the proposed workflow for consuming this action, maybe this should be required? If the consumer doesn't want to publish to npm, they can just not use this action.

File: action.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: 'Publish to npm'
```
Comment (mcmire) [inline action.yml:1]: Do our actions get published to npm? I always thought they just lived on GitHub.

Comment (rickycodes) [reply action.yml:1]: nope! they just live on GitHub.

**Pull request merged**